src/frequenz/client/dispatch/_client.py

```diff
@@ -0,0 +1,234 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Dispatch API client for Python."""
+from datetime import datetime, timedelta
+from typing import Any, Awaitable
+
+import grpc
+from frequenz.api.dispatch.v1 import dispatch_pb2_grpc
+
+# pylint: disable=no-name-in-module
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    DispatchDeleteRequest,
+    DispatchFilter,
+    DispatchGetRequest,
+    DispatchList,
+    DispatchListRequest,
+    DispatchUpdateRequest,
+)
+from frequenz.api.dispatch.v1.dispatch_pb2 import (
+    TimeIntervalFilter as PBTimeIntervalFilter,
+)
+from google.protobuf.timestamp_pb2 import Timestamp
+
+from ._types import (
+    ComponentSelector,
+    Dispatch,
+    DispatchCreateRequest,
+    RecurrenceRule,
+    _component_selector_to_protobuf,
+)
+
+# pylint: enable=no-name-in-module
+
+
+class Client:
+    """Dispatch API client."""
+
+    def __init__(self, grpc_channel: grpc.aio.Channel, svc_addr: str) -> None:
+        """Initialize the client.
+
+        Args:
+            grpc_channel: gRPC channel to use for communication with the API.
+            svc_addr: Address of the service to connect to.
+        """
+        self._svc_addr = svc_addr
+        self._stub = dispatch_pb2_grpc.MicrogridDispatchServiceStub(grpc_channel)
+
+    # pylint: disable=too-many-arguments, too-many-locals
+    async def list(
+        self,
+        microgrid_id: int,
+        component_selectors: list[ComponentSelector] | None = None,
+        start_from: datetime | None = None,
+        start_to: datetime | None = None,
+        end_from: datetime | None = None,
+        end_to: datetime | None = None,
+        is_active: bool | None = None,
+        is_dry_run: bool | None = None,
+    ) -> list[Dispatch]:
```

Comment:
I think this should return an async iterator instead.
```suggestion
    ) -> AsyncIterator[Dispatch]:
```

So users can just do:

```py
async for dispatch in client.list():
    ...
```

Once the return type is `AsyncIterator`, you can just `yield` each value in the function body. This will be particularly important when we add pagination, so users don't need to wait (and hold in memory) all the pages before starting to process dispatches.